### ISSUE TITLE
Elixir 1.6 produces compilation warnings

### ISSUE THREAD

**Issue by halfdan** [rrrene/credo#496]
### Environment

* Credo version (`mix credo -v`): 0.8.10
* Erlang/Elixir version (`elixir -v`): 1.6.0
* Operating system: MacOS 10.13.2

### What were you trying to do?

We're running warnings as errors and credo is one of a number of packages that produces several warnings on Elixir 1.6 preventing us from upgrading to 1.6.

### Expected outcome

Compilation succeeds without any warnings.

### Actual outcome

```
==> credo
Compiling 161 files (.ex)
warning: this clause cannot match because a previous clause at line 165 always matches
  lib/credo/cli/output/summary.ex:165

warning: module attribute @lint was set but never used
  lib/credo/check/find_lint_attributes.ex:23
```

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by xtian** [comment]
> We're running warnings as errors and credo is one of a number of packages that produces several warnings on Elixir 1.6 preventing us from upgrading to 1.6.

@halfdan You might consider configuring your CI build to run `mix deps.compile` before `mix compile --warnings-as-errors`

**Comment by rrrene** [comment]
Thx for reporting this! :+1: It should be be fixed in the latest version of Credo.

If it is not, please feel free to re-open this issue!
